### PR TITLE
Messages pagination v4

### PR DESCRIPTION
**Pull request by ValentinaHotsa**
In this branch I add component DateSeparator for separating between date in chat. And add pagination for list of messages

### CODE REVIEW

File: src/components/MessageList/MessageList.jsx [hunk 1]
```diff
@@ -1,15 +1,13 @@
-import { useEffect, useRef } from 'react';
+import { useEffect } from 'react';
+import { isSameDay } from 'date-fns';
+
 import PropTypes from 'prop-types';
 import MessageItem from '@/components/MessageItem/MessageItem';
 import { MESSAGE_TYPES } from '@/constants/messageTypes.js';
 import { MessageListStyled } from './MessageListStyled.js';
+import DateSeparator from '../DateSeparator/DateSeparator.jsx';
```
Comment (ReginaVarybrus) [inline src/components/MessageList/MessageList.jsx]: Lets rename a path to '@/components/DateSeparator/DateSeparator.jsx'

File: src/components/DateSeparator/DateSeparatorStyles.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  display: flex;
+  align-items: center;
+  margin: 16px 0;
+  color: var(--color-grey-9);
+  @media ${device.tablet} {
+    margin: 24px 0;
+  }
+`;
+
+export const Line1 = styled.div`
```
Comment (ReginaVarybrus) [inline src/components/DateSeparator/DateSeparatorStyles.js:14]: Is it possible to do it with pseudoelements :after, :before. I'm not sure it`s better))

**Review by ReginaVarybrus — CHANGES_REQUESTED** [review]

File: src/components/MessageList/MessageList.jsx (hunk 1 above)
Comment (ValentinaHotsa) [reply src/components/MessageList/MessageList.jsx]: done

File: src/components/DateSeparator/DateSeparatorStyles.js (hunk 2 above)
Comment (ValentinaHotsa) [reply src/components/DateSeparator/DateSeparatorStyles.js:14]: with pseudoelements I had problems with styles, so I took the simpler route))

File: src/components/Chat/Chat.jsx [hunk 3]
```diff
@@ -46,9 +71,182 @@ const Chat = ({
   const isChatVisible = context?.isChatVisible;
   const setIsChatVisible = context?.setIsChatVisible;
 
+  const markAsRead = async chatId => {
+    try {
+      if (messages.length > 0) {
+        const lastMessageId = messages[messages.length - 1]?.id;
+        if (lastMessageId) {
+          await axiosClient.patch(`/chats/${chatId}/messages/last-read`, {
```
Comment (LiashenkoAndrey) [inline src/components/Chat/Chat.jsx]: hard coded string

Comment (ReginaVarybrus) [inline src/components/Chat/Chat.jsx]: Let`s use endpoints from constants.js

File: src/components/Chat/Chat.jsx [hunk 4]
```diff
@@ -46,9 +71,182 @@ const Chat = ({
...
+      }
+    } catch (error) {
+      console.error('Error updating last read message:', error);
+    }
+  };
+
+  const fetchPublicMessages = async (pageNumber = 0) => {
+    setIsFetchingMore(true);
+    try {
+      const response = await axiosClient.get(`/chats/${id}/messages`, {
```
Comment (ReginaVarybrus) [inline src/components/Chat/Chat.jsx]: The same

File: src/components/Chat/Chat.jsx [hunk 5]
```diff
@@ -46,9 +71,182 @@ const Chat = ({
...
+      console.error('Error fetching messages:', error);
+    } finally {
+      setIsFetchingMore(false);
+    }
+  };
+
+  const fetchMessages = async (pageNumber = 0) => {
+    setIsFetchingMore(true);
+    try {
+      const response = await axiosClient.get(`/chats/${id}/messages/read`, {
```
Comment (ReginaVarybrus) [inline src/components/Chat/Chat.jsx]: The same

File: src/components/Chat/Chat.jsx [hunk 6]
```diff
@@ -46,9 +71,182 @@ const Chat = ({
...
+    } catch (error) {
+      console.error('Error fetching messages:', error);
+    } finally {
+      setIsFetchingMore(false);
+    }
+  };
+
+  const fetchUnreadMessages = async (pageNumber = 0) => {
+    try {
+      const response = await axiosClient.get(`/chats/${id}/messages/unread`, {
```
Comment (ReginaVarybrus) [inline src/components/Chat/Chat.jsx]: The same? and maybe we should move all this functions to separate file in order to clean this component.

**Review by ReginaVarybrus — CHANGES_REQUESTED** [review]

File: src/components/Chat/Chat.jsx (hunk 3 above)
Comment (ValentinaHotsa) [reply src/components/Chat/Chat.jsx]: done

File: src/components/Chat/Chat.jsx (hunk 4 above)
Comment (ValentinaHotsa) [reply src/components/Chat/Chat.jsx]: done

File: src/components/Chat/Chat.jsx (hunk 5 above)
Comment (ValentinaHotsa) [reply src/components/Chat/Chat.jsx]: done

File: src/components/Chat/Chat.jsx (hunk 6 above)
Comment (ValentinaHotsa) [reply src/components/Chat/Chat.jsx]: done
yes, I thought about it

**Review by ReginaVarybrus — APPROVED** [review]

**Pull request merged**